assembly/__tests__/runtime/main.ts

```diff
@@ -170,6 +171,25 @@ export function storageGenericGetSetRoundtripTest(): void {
   );
 }
 
+export function storageRawRoundtripTest(): void {
+  logging.log("storageRawRoundtripTest");
+
+  storage.set<string>("message1", "Hello World");
+  storage.read_raw("message1", 0);
+  const regGet = util.read_register_string(0);
+  assert(
+    regGet == "Hello World",
+    "Incorrect data value for reading string from registry"
+  );
+
+  storage.write_raw("message2", 0);
+  const stringGet = storage.get<string>("message2");
+  assert(
+    stringGet == "Hello World",
```

Comment:
@willemneal Ok I don't know if I'm going crazy or not, but this test case keeps failing. I can't figure out why -- the test case above passes, so it means we're properly writing to the register, so if it fails here, it can only mean that the storage isn't being properly written with the register's value, which is impossible -- I've double-checked the `write_raw` function a few times and I don't really seem to see why it's not working as intended. `stringGet` here is an empty string.